service/history/historyBuilder.go

```diff
@@ -119,10 +119,8 @@ func (b *historyBuilder) AddDecisionTaskTimedOutEvent(scheduleEventID int64,
 	return b.addEventToHistory(event)
 }
 
-func (b *historyBuilder) AddDecisionTaskFailedEvent(scheduleEventID int64, startedEventID int64,
-	cause workflow.DecisionTaskFailedCause, details []byte, identity string) *workflow.HistoryEvent {
-	event := b.newDecisionTaskFailedEvent(scheduleEventID, startedEventID, cause, details, identity)
-
+func (b *historyBuilder) AddDecisionTaskFailedEvent(attr workflow.DecisionTaskFailedEventAttributes) *workflow.HistoryEvent {
```

Comment:
It kind of breaks the existing pattern.  This API takes the necessary fields as input parameters and then build the corresponding *EventAttributes within this implementation.  Can you just add the new fields as input parameters for now.

Reply:
OK